app/js/enhancedDate.js

```diff
@@ -1,5 +1,63 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function() {
+	
+	'use strict'
 
+	var myDate = null;
+	var dayNames = ['Monday','Tuesday','Wednesday','Thursday','Friday','Saturday','Sunday'];
+	var monthNames = ['January','February','March','April','May','June','July','August','September','October','November','December'];
+
+	function setDate(date) {
+		if (date instanceof Date || typeof date ==='number') {
+			myDate = new Date(date);
+		} else {
+			myDate = new Date();
+		}
+	}
+
+	function getDate(returnObj) {
+		if (returnObj) {
```

Comment:
The one part missing from all of the enhancedDate methods, save for setDate, is a check to see if myDate had been set first, and if not, then setDate() called to set it before doing anything else with myDate.